library/core/src/cell.rs

```diff
@@ -2042,6 +2042,7 @@ impl<T> UnsafeCell<T> {
     #[rustc_const_stable(feature = "const_unsafe_cell_new", since = "1.32.0")]
     #[inline(always)]
     pub const fn new(value: T) -> UnsafeCell<T> {
+        #[allow(large_assignments)]
         UnsafeCell { value }
     }
```

Comment:
Is this actually reported while compiling libcore? Or is this from a use of `Cell::new` with large types?

Reply:
It must be the latter. Unfortunately the diagnostics does not tell from where. For the `black_box()` case the diagnostic looks like this:

```
error: moving 65536 bytes
   --> /home/martin/src/rust/library/core/src/hint.rs:293:5
    |
293 |     crate::intrinsics::black_box(dummy)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ value moved from here
    |
    = note: The current maximum size is 4096, but it can be customized with the move_size_limit attribute: `#![move_size_limit = "..."]`
    = note: `-D large-assignments` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(large_assignments)]`

error: moving 131072 bytes
   --> /home/martin/src/rust/library/core/src/hint.rs:293:5
    |
293 |     crate::intrinsics::black_box(dummy)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ value moved from here
    |
    = note: The current maximum size is 4096, but it can be customized with the move_size_limit attribute: `#![move_size_limit = "..."]`

error: could not compile `alloc` (bench "collectionsbenches") due to 2 previous errors
```
I added a task that we should add to the tracking issue that we need to improve diagnostics.